src/apply/value/setValue.js

```diff
@@ -1,9 +1,8 @@
 const Y = require('yjs');
 
 const setValue = (doc, op)=>{
-    if(!doc.hasOwnProperty('data')){
-        doc.set("data", new Y.Map())
-    }
+    doc.set("data", new Y.Map())
```

Comment:
I'll fully admit I don't understand the larger code flow here, but not doing this conditionally feels like it could overwrite existing content, which would presumably be bad.

Per the other (longer) comment thread, I wonder if the preferred path here is to call getMap() on the top-level Y.Doc?

Reply:
I was informed that the data values when they are updated the whole data structure are transmitted, and as the slate doc shows [https://github.com/hugoai/slate/blob/master/docs/reference/slate/data.md](https://github.com/hugoai/slate/blob/master/docs/reference/slate/data.md) the Data is immutable so it would be impossible to change one field, the whole Data need to be rewritten. 

Is that right?

Reply:
Overwriting the whole `data` content is the same behavior the operation has in the Slate-only side of things, we had to make sure that whenever we update it we copy the current content and add to it, so I think that's fine.

We could, however, reverse the order of how things are done here to avoid multiple Yjs events? Something like this:

```js
const newData = new Y.Map();
// Update new data here with the with operation data
doc.set('data', newData);
```
Then, Yjs would probably interpret it as a single mutation to the main doc and dispatch a single event.

Reply:
Yeah, I think that could be better, change the whole newData then push it to the doc should save us multiple Y.js propagations. Gonna implement and see if everything works.

Reply:
re: "dispatch a single event" -- while I like this approach, I'm not certain it makes a big difference at the Yjs level, where I believe things are operating more in terms of diffs than events.